### PR TITLE
TST: fix deprecation warning in test 

### PR DESCRIPTION
**Pull request by attack68**
deprecation warning regarding ambiguous boolean of empty array (numpy)

### CODE REVIEW

File: pandas/tests/io/formats/style/test_style.py [hunk 1]
```diff
@@ -319,7 +319,12 @@ def test_clear(mi_styler_comp):
     # tests vars are not same vals on obj and clean copy before clear (except for excl)
     for attr in [a for a in styler.__dict__ if not (callable(a) or a in excl)]:
         res = getattr(styler, attr) == getattr(clean_copy, attr)
-        assert not (all(res) if (hasattr(res, "__iter__") and len(res) > 0) else res)
+        if hasattr(res, "__iter__") and len(res) > 0:
+            assert not (all(res))  # some element in iterable differs
+        elif hasattr(res, "__iter__") and len(res) == 0:
+            pass  # empty array
+        else:
+            assert not (res)  # explicit var differs
```
Comment (jbrockmendel) [inline pandas/tests/io/formats/style/test_style.py:327]: parens might be unnecessary?  same on L323

File: pandas/tests/io/formats/style/test_style.py [hunk 2]
```diff
@@ -319,7 +319,12 @@ def test_clear(mi_styler_comp):
     # tests vars are not same vals on obj and clean copy before clear (except for excl)
     for attr in [a for a in styler.__dict__ if not (callable(a) or a in excl)]:
         res = getattr(styler, attr) == getattr(clean_copy, attr)
-        assert not (all(res) if (hasattr(res, "__iter__") and len(res) > 0) else res)
+        if hasattr(res, "__iter__") and len(res) > 0:
+            assert not all(res)  # some element in iterable differs
+        elif hasattr(res, "__iter__") and len(res) == 0:
+            pass  # empty array
+        else:
+            assert not res  # explicit var differs
```
Comment (jbrockmendel) [inline pandas/tests/io/formats/style/test_style.py:327]: would this whole thing be simpler as `not np.all(res)`?

Comment (attack68) [reply pandas/tests/io/formats/style/test_style.py:327]: tried it. doesn't work cause its looping over different types of attributes, lists, dicts, arrays, and scalars. so the above is, possibly, the most general way of catching the cases.

Comment (jbrockmendel) [reply pandas/tests/io/formats/style/test_style.py:327]: makes sense thanks

**Review by jbrockmendel — APPROVED** [review]
LGTM

**Pull request merged**